src/App.tsx

```diff
@@ -1,34 +1,51 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import './App.css'
+import { useEffect } from "react";
+import { Routes, Route, Navigate } from "react-router-dom";
+import "./App.css";
+import Navbar from "./components/navbar/navbar";
+import { BooksPage } from "./domain/books/BooksPage";
+import { CartPage } from "./domain/cart/CartPage";
+import {
+  getBooks
+} from "./domain/books/booksSlice";
+import { useAppDispatch, useAppSelector } from "./store/store";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const dispatch = useAppDispatch();
+  const { cartItems } = useAppSelector((state) => state.cart);
+  const { books, loading } = useAppSelector((state) => state.books);
+
+  useEffect(() => {
+    dispatch(getBooks());
+  }, []);
+
 
   return (
-    <div className="App">
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src="/vite.svg" className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://reactjs.org" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+    <div className='main'>
+      <Navbar cartItems={cartItems} />
+      <hr></hr>
+      <Routes>
+        <Route path='/' element={<Navigate to='/home' />} />
+        <Route
+          path='/home'
+          element={
+            <BooksPage
+              books={books}
+              loading={loading}
+            />
+          }
+        />
+        <Route
+          path='/cart'
+          element={
+            <CartPage
+              cartItems={cartItems}
+            />
+          }
+        />
+        <Route path='*' element={<Navigate to='/home' />} />
+      </Routes>
```

Comment:
As stated earlier, all the routes should be declared in src/router/Router.tsx file and imported here

Reply:
Also, do not pass any props to an element declared in a route.
In each page, i.e. BooksPage & CartPage, get the required state from the store using useSelector